### PR TITLE
CSP-1209 apprentice aan upgraded to Net8 and shared outer moved into …

### PR DESCRIPTION
**Pull request by MarkFCain**
…codebase

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -552,3 +552,24 @@ You are able to run the API by doing the following:
 
 Starting the API will then show the swagger definition with the available operations. This outer API is used to provide data for the FindApprenticeshipJobs functions app:
 [das-findapprenticeship-jobs](https://github.com/SkillsFundingAgency/das-findapprenticeship-jobs)
+
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply README.md:555]: ok, I'll pick this up

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MyApprenticeships/Queries/GetMyApprenticeship/GetMyApprenticeshipQueryHandler.cs [hunk 2]
```diff
@@ -1,64 +1,65 @@
...
-    private async Task<GetMyApprenticeshipQueryResult?> ConvertResponseToResultWithTrainingCourse(GetMyApprenticeshipResponse myApprenticeshipsResponse)
+    private async Task<GetMyApprenticeshipQueryResult?> ConvertResponseToResultWithTrainingCourse(GetMyApprenticeshipResponse myApprenticeshipsResponse, CancellationToken cancellationToken)
     {
         var result = (GetMyApprenticeshipQueryResult)myApprenticeshipsResponse;
 
         if (myApprenticeshipsResponse.StandardUId != null)
         {
-            var standard = await _coursesApiClient.Get<GetStandardResponse>(
-                new GetStandardRequest(myApprenticeshipsResponse.StandardUId));
+            // var standard = await _coursesApiClient.Get<GetStandardResponse>(
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MyApprenticeships/Queries/GetMyApprenticeship/GetMyApprenticeshipQueryHandler.cs:40]: removed

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MyApprenticeships/Queries/GetMyApprenticeship/GetMyApprenticeshipQueryHandler.cs [hunk 3]
```diff
@@ -1,64 +1,65 @@
...
             {
-                result.TrainingCourse = standard;
+                result.TrainingCourse = standard.GetContent();
             }
         }
         else if (myApprenticeshipsResponse.TrainingCode != null)
         {
-            var framework = await _coursesApiClient.Get<GetFrameworkResponse>(
-                new GetFrameworkRequest(myApprenticeshipsResponse.TrainingCode));
+            // var framework = await _coursesApiClient.Get<GetFrameworkResponse>(
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MyApprenticeships/Queries/GetMyApprenticeship/GetMyApprenticeshipQueryHandler.cs:53]: removed

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Configuration/AanHubApiConfiguration.cs
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Configuration/AanHubApiConfiguration.cs:1]: When I looked into this, there were several configurations no longer needed, this and those other ones are now removed

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/AppStart/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -90,4 +110,7 @@ private static void AddConfigurationOptions(IServiceCollection services, IConfig
         services.Configure<CoursesApiConfiguration>(configuration.GetSection(nameof(CoursesApiConfiguration)));
         services.AddSingleton(c => c.GetService<IOptions<CoursesApiConfiguration>>()!.Value);
     }
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/AppStart/ServiceCollectionExtensions.cs:112]: done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/Configuration/ApprenticeAccountsApiConfiguration.cs
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/Configuration/ApprenticeAccountsApiConfiguration.cs:1]: done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/Configuration/CoursesApiConfiguration.cs
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/Configuration/CoursesApiConfiguration.cs:1]: done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/Configuration/LocationApiConfiguration.cs
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/Configuration/LocationApiConfiguration.cs:1]: done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MyApprenticeships/Commands/CreateMyApprenticeship/CreateMyApprenticeshipCommandHandler.cs [hunk 5]
```diff
@@ -1,28 +1,19 @@
...
+    public async Task<string> Handle(CreateMyApprenticeshipCommand command, CancellationToken cancellationToken)
     {
-        PostMyApprenticeshipRequest apiRequest = new(command.ApprenticeId) { Data = command };
-
-        var response = await _apprenticeAccountsApiClient.PostWithResponseCode<object>(apiRequest, false);
-
-        if (response.StatusCode == HttpStatusCode.Created) return Unit.Value;
-
-        throw new InvalidOperationException($"An attempt to create MyApprenticeship for ApprenticeId: {command.ApprenticeId}, came back with unsuccessful response status: {response.StatusCode}");
+        return await _apprenticeAccountsApiClient.PostMyApprenticeship(command.ApprenticeId, command, cancellationToken);
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MyApprenticeships/Commands/CreateMyApprenticeship/CreateMyApprenticeshipCommandHandler.cs:17]: Just a status code, changing to suit that.

[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MyApprenticeships/Commands/CreateMyApprenticeship/CreateMyApprenticeshipCommandHandler.cs:17]: this is wrong, it needs to deal with http response, changing to suit

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/IApprenticeAccountsApiClient.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    Task<Response<GetApprenticeAccountQueryResult>> GetApprentice([Path] Guid apprenticeId, CancellationToken cancellationToken);
+
+
+    [Get("apprentices/{apprenticeId}/MyApprenticeship")]
+    [AllowAnyStatusCode]
+    Task<Response<GetMyApprenticeshipResponse>> GetMyApprenticeship([Path] Guid apprenticeId, CancellationToken cancellationToken);
+
+    [Post("apprentices/{apprenticeId}/MyApprenticeship")]
+    [AllowAnyStatusCode]
+    Task<string> PostMyApprenticeship(
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Infrastructure/IApprenticeAccountsApiClient.cs:20]: http response. Fixed

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Locations/Queries/GetAddresses/GetAddressesQueryHandler.cs [hunk 7]
```diff
@@ -1,26 +1,25 @@
...
     public const double MaximumMatch = 1;
     public const string PostcodeRegex = @"^[a-zA-z]{1,2}\d[a-zA-z\d]?\s*\d[a-zA-Z]{2}$";
 
-    public GetAddressesQueryHandler(ILocationApiClient<LocationApiConfiguration> locationApiClient) => _locationApiClient = locationApiClient;
+
+    public GetAddressesQueryHandler(ILocationApiClient locationApiClient) => _locationApiClient = locationApiClient;
 
     public async Task<GetAddressesQueryResult> Handle(GetAddressesQuery request, CancellationToken cancellationToken)
     {
-        var isFullPostcode = Regex.IsMatch(request.Query, PostcodeRegex);
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/Locations/Queries/GetAddresses/GetAddressesQueryHandler.cs:21]: I did this and pushed, no idea why it's not showing 'outdated' here

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarkFCain) [reply README.md:555]: I have done the readme changes as a separate commit, so it's easier to review the code changes, distinct from readme

**Review by ManishAgarwalDfE — APPROVED** [review]

**Review by ManishAgarwalDfE — APPROVED** [review]

**Pull request merged**